lib/apiservers/engine/backends/container.go

```diff
@@ -1543,6 +1546,48 @@ func ContainerSignal(containerID string, sig uint64) error {
 	return nil
 }
 
+func getPortInformation(t *models.ContainerInfo, names []string) []types.Port {
+	var ports []types.Port
+	if len(names) == 0 {
+		log.Errorf("Couldn't find container while looking up port bindings; No container names provided")
+		return ports
+	}
+
+	var container *viccontainer.VicContainer
+	container = cache.ContainerCache().GetContainer(names[0])
+	if container == nil {
+		if match := regexp.MustCompile(`[\w_]+`).FindString(names[0]); match != "" {
+			container = cache.ContainerCache().GetContainer(match)
+		} else {
+			log.Errorf("Could not find container based on incorrectly formatted name")
+			return ports
+		}
+	}
+
+	portBindings := container.HostConfig.PortBindings
+
+	for _, jsonPort := range t.ContainerConfig.Ports {
+		port := new(types.Port)
+		err := json.Unmarshal([]byte(jsonPort), port)
+		if err != nil {
```

Comment:
Yes, there's a `continue` on the next line. We just report the error and move on. It's not fatal, to the rest of the operation. It may not even be fatal to the rest of the ports in the slice being iterated over.